shoes-core/lib/shoes/line.rb

```diff
@@ -19,6 +21,26 @@ def create_dimensions(point_a, point_b)
       style[:y2] = point_b.y
     end
 
+    # Check out http://math.stackexchange.com/questions/60070/checking-whether-a-point-lies-on-a-wide-line-segment
+    # for explanations how the algorithm works
+    def in_bounds?(x, y)
+      # c is (x, y)
+      left_most, right_most = point_a.x < point_b.x ? [point_a, point_b] : [point_b, point_a]
+      vec_left_c = Vector[left_most.x - x, left_most.y - y]
+      vec_left_right = Vector[left_most.x - right_most.x, left_most.y - right_most.y]
+
+      boldness = style[:strokewidth].to_i / 2
+      vec_left_c_dot_vec_left_right = vec_left_c.inner_product(vec_left_right)
+      vec_left_right_dot_vec_left_right = vec_left_right.inner_product(vec_left_right)
+
+      if vec_left_c_dot_vec_left_right.between?(0, vec_left_right_dot_vec_left_right)
+        vec_left_c_dot_vec_left_c = vec_left_c.inner_product(vec_left_c)
```

Comment:
Hm, yes, they read funny... I don't know. Math is too long ago for me to know if there is a good name for the product (like what it essentially means). Maybe leaving out the vec? `left_right` makes it a bit clear that it is a vector. You are right, right now the names are a bit too long.

Sorry for the name nit picking, it's a pet peeve of mine :)